base/steer/FairRunAna.cxx

```diff
@@ -503,6 +507,8 @@ void FairRunAna::Run(Long64_t entry)
 {
     UInt_t tmpId = 0;
     fRootManager->ReadEvent(entry);
+    if(!fEvtHeader)
+        fEvtHeader = GetEventHeader();
     tmpId = fEvtHeader->GetRunId();
```

Comment:
I don't really like that we have to check each instance of `->GetRunId()` for fEvtHeader being set…

What do you think about [this](https://github.com/ChristianTackeGSI/FairRoot/commit/fa238b90f8a27edeae66c34056ef900ffb50cf34) (you might want to cherry-pick it into your branch?)